test/nix/initramfs.nix

```diff
@@ -26,6 +30,24 @@ in stdenvNoCC.mkDerivation {
 
     cp -r ${etc}/* $out/etc/
 
+    RESOLV_CONF_FILE="$out/etc/resolv.conf"
```

Comment:
Please move `resolve.conf` logic to a separate file or Nix expression. It should NOT be added directly to this Nix expression, making it too large to read.